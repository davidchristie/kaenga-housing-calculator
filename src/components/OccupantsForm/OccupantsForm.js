import React, { Component, PropTypes } from 'react'
import { Panel } from 'react-bootstrap'
import { Field } from 'redux-form'

import AdultsField from './AdultsField'
import ChildrenField from './ChildrenField'
import TeenagersField from './TeenagersField'

class OccupantsForm extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel header='Occupants'>
          <Field component={AdultsField} name='adults' />
          <Field component={TeenagersField} name='teenagers' />
          <Field component={ChildrenField} name='children' />
        </Panel>
      </form>
    )
  }
}

OccupantsForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default OccupantsForm
