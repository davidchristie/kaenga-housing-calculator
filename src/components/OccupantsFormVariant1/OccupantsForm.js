import React, { Component, PropTypes } from 'react'
import { Button, Panel } from 'react-bootstrap'
import { FieldArray, reduxForm } from 'redux-form'

import OccupantListField from './OccupantListField'

class OccupantsForm extends Component {
  render () {
    const { handleSubmit, pristine, reset, submitting } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel bsStyle='warning' header='Occupants (Variant #1)'>
          <FieldArray
            component={OccupantListField}
            name='occupants'
          />
          <Button
            disabled={pristine || submitting}
            onClick={reset}
          >
            Clear Occupants
          </Button>
        </Panel>
      </form>
    )
  }
}

OccupantsForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default reduxForm({
  form: 'occupantsVariant1'
})(OccupantsForm)
