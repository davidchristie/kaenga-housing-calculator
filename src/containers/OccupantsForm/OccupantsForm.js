import { reduxForm } from 'redux-form'

import OccupantsForm from '../../components/OccupantsForm'

export default reduxForm({
  form: 'occupants'
})(OccupantsForm)
