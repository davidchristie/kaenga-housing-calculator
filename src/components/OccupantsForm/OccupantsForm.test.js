import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import OccupantsForm from './OccupantsForm'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<OccupantsForm />)).toMatchSnapshot()
})
