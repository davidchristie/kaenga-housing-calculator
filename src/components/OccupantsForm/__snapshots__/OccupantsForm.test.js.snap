exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Field
    component={[Function]}
    name="adults" />
  <Field
    component={[Function]}
    name="teenagers" />
  <Field
    component={[Function]}
    name="children" />
</form>
`;
